packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -314,48 +309,29 @@ export class SObjectDescribe {
       }
       return Promise.resolve(fetchedObjects);
     } catch (error) {
-      const xhrResponse: XHRResponse = error;
-      return Promise.reject(xhrResponse.responseText);
+      const errorMsg = error.hasOwnProperty('responseText')
+        ? error.responseText
+        : error.message;
+      return Promise.reject(errorMsg);
     }
   }
 
-  // get the token and url by calling the org - short term, should really be able to get it from the sfdx project
-  // also set the proper target apiVersion
-  private async setupConnection(projectPath: string, username?: string) {
-    if (!this.accessToken) {
-      let orgInfo: any;
-      const builder = new SfdxCommandBuilder().withArg('force:org:display');
-      if (username) {
-        builder.args.push('--targetusername', username);
-      }
-      const command = builder.withJson().build();
-      const execution = new CliCommandExecutor(command, {
-        cwd: projectPath
-      }).execute();
-      const cmdOutput = new CommandOutput();
-      const result = await cmdOutput.getCmdResult(execution);
-      orgInfo = JSON.parse(result).result;
-      this.accessToken = orgInfo.accessToken;
-      this.instanceUrl = orgInfo.instanceUrl;
-    }
-    if (!this.targetVersion) {
-      this.targetVersion = await this.getTargetApiVersion(projectPath);
+  public async getConnectionData(projectPath: string) {
+    try {
+      const username = await ConfigUtil.getUsername(projectPath);
+      const authInfo = await AuthInfo.create({ username });
+      const opts = authInfo.getConnectionOptions();
+      this.accessToken = opts.accessToken;
+      this.instanceUrl = opts.instanceUrl;
+    } catch (err) {
+      const error = new Error();
+      error.name = 'Authentication Error';
+      error.message = err.message;
+      throw error;
```

Comment:
Can we log when there is a failure to authenticate?

Reply:
These errors bubble up and get logged by the error handling code in `forceGenerateFauxClasses.ts`